src/cpp/session/SessionMain.cpp

```diff
@@ -3243,9 +3260,12 @@ int main (int argc, char * const argv[])
       error = workingDir.makeCurrentPath();
       if (error)
          return sessionExitFailure(error, ERROR_LOCATION);
-         
-      // start http connection listener
-      error = startHttpConnectionListener();
+      
+      // start http connection listener and wait for the port to be ready,
+      // if needed. For instance, when the rsession restarts but takes the
+      // port a few ms to become available
+      error = retry(boost::bind(startHttpConnectionListener),
```

Comment:
You *might* be able to just pass startHttpConnectionListener without the boost::bind since it matches the required signature exactly.